src/cbmc/cbmc_parse_options.cpp

```diff
@@ -835,16 +835,21 @@ bool cbmc_parse_optionst::process_goto_program(
     // Similar removal of java nondet statements:
     // TODO Should really get this from java_bytecode_language somehow, but we
     // don't have an instance of that here.
-    const auto max_nondet_array_length=
+    const size_t max_nondet_array_length=
       cmdline.isset("java-max-input-array-length")
-        ? std::stoi(cmdline.get_value("java-max-input-array-length"))
+        ? std::stoul(cmdline.get_value("java-max-input-array-length"))
         : MAX_NONDET_ARRAY_LENGTH_DEFAULT;
+    const size_t max_nondet_tree_depth=
```

Comment:
Does this only limit recursive datatypes (so non-recursive types are fully elaborated as before)? If so rename it to indicate that